src/sentry/tasks/backfill_seer_grouping_records.py

```diff
@@ -0,0 +1,335 @@
+                ),
+                Condition(Column("timestamp", entity=events_entity), Op.LT, time_now),
+            ],
+            orderby=[OrderBy(Column("group_id"), Direction.ASC)],
+        )
+
+        request = Request(
+            dataset=Dataset.Events.value,
+            app_id=Referrer.GROUPING_RECORDS_BACKFILL_REFERRER.value,
+            query=query,
+            tenant_ids={
+                "referrer": Referrer.GROUPING_RECORDS_BACKFILL_REFERRER.value,
+                "cross_org_query": 1,
+            },
+        )
+
+        with metrics.timer(f"{BACKFILL_NAME}.bulk_snuba_queries", sample_rate=1.0):
+            result = bulk_snuba_queries(
+                [request], referrer=Referrer.GROUPING_RECORDS_BACKFILL_REFERRER.value
+            )
+
+        if result and result[0].get("data"):
+            rows: list[GroupEventRow] = result[0]["data"]
+            group_hashes = GroupHash.objects.filter(
+                project_id=project_id, group_id__in=group_id_batch
+            ).distinct("group_id")
+            group_hashes_dict = {
+                group_hash.group_id: group_hash.hash for group_hash in group_hashes
+            }
+            data = lookup_group_data_stacktrace_bulk_with_fallback(
+                project, rows, group_id_message_batch, group_hashes_dict
+            )
+            remove_grouping_record_table_init = features.has(
+                "projects:similarity-embeddings-remove-seer-grouping-record-table-init", project
+            )
+            response = post_bulk_grouping_records(
+                CreateGroupingRecordsRequest(
+                    group_id_list=group_id_batch,
+                    data=data["data"],
+                    stacktrace_list=data["stacktrace_list"],
+                    remove_grouping_record_table_init=remove_grouping_record_table_init,
+                )
+            )
+            if response["success"]:
+                redis_client.set(
+                    f"{LAST_PROCESSED_REDIS_KEY}", group_id_batch[-1], ex=60 * 60 * 24 * 7
+                )
+                groups = Group.objects.filter(project_id=project_id, id__in=group_id_batch)
+                for group in groups:
+                    if group.data.get("metadata"):
+                        group.data["metadata"]["embeddings_info"] = {
+                            "nn_model_version": 1,
+                            "group_hash": json.dumps([group_hashes_dict[group.id]]),
+                        }
+                    else:
+                        group.data["metadata"] = {
+                            "embeddings_info": {
+                                "nn_model_version": 1,
+                                "group_hash": json.dumps([group_hashes_dict[group.id]]),
+                            }
+                        }
+                num_groups_records_created += Group.objects.bulk_update(groups, ["data"])
+            time.sleep(ITERATOR_CHUNK * SEER_BACKFILL_DELAY_PER_RECORD)
```

Comment:
How long are you expecting this task to take? If it's pretty long you might hit a timeout.

One option here is to refire the task and pass the id you just completed and continue on